administrator/components/com_content/controllers/article.php

```diff
@@ -154,8 +155,35 @@ public function batch($model = null)
 	 *
 	 * @since	3.1
 	 */
-	protected function postSaveHook(JModelLegacy $model, $validData = array())
+	protected function postSaveHook(JModelLegacy $model, $validData)
 	{
+		JModelLegacy::addIncludePath(JPATH_ADMINISTRATOR . '/components/com_menus/models', 'MenusModel');
+		$itemModel = JModelAdmin::getInstance('Item', 'MenusModel');
+		$itemModel->addTablePath(JPATH_ADMINISTRATOR . '/components/com_menus/tables');
+
+		$componentId = JComponentHelper::getComponent('com_content')->id;
+
+		JLoader::register('ContentHelper', JPATH_ADMINISTRATOR . '/components/com_content/helpers/content.php');
+		$articleId = ContentHelper::getarticleid($validData['title']);
+
+		$vaData['id'] = 0;
+		$vaData['menutype'] = $validData['menulink']['menutype'];
+		$vaData['title'] = $validData['menulink']['menutitle'];
+		$vaData['alias'] = $validData['menulink']['menualias'];
+		$vaData['link'] = 'index.php?option=com_content&view=article&id=' . $articleId;
+		$vaData['type'] = 'component';
+		$vaData['published'] = 1;
+		$vaData['parent_id'] = 1;
+		$vaData['level'] = 1;
+		$vaData['component_id'] = $componentId;
```

Comment:
Instead of creating new variable you can directly use it. No needs to create variable unless you are using it at more than one place in scope.

```php
$vaData['component_id'] = JComponentHelper::getComponent('com_content')->id;
```